src/Servers/Kestrel/Core/test/HttpHeadersTests.cs

```diff
@@ -287,4 +287,30 @@ private static void InvalidContentLengthsRejectedImpl(HttpHeaders httpHeaders)
         Assert.Null(httpHeaders.ContentLength);
         Assert.False(httpHeaders.ContentLength.HasValue);
     }
+
+    [Theory]
+    [InlineData("foo", "bar", true)]
+    [InlineData("foo", "ba\tr", true)]
+    [InlineData("foo", "ba r", true)]
+    [InlineData("foo", "ba\rr", false)]
+    [InlineData("foo", "ba\r\nr", false)]
+    [InlineData("foo", "ba\nr", false)]
+    [InlineData("fo\to", "bar", false)]
+    public void TestParseCustomHeaderValues(string headerName, string headerValue, bool expectSuccess)
```

Comment:
I would have split it in two tests, one for headers and one for value. Or is that applying the same logic as another test?

Reply:
Yea, that's cleaner. I'll change it.